File: docs/snippets/csharp/user-message_main.cs
// using Microsoft.Agents.Xml.Generated.Models;
// using Microsoft.Agents.Protocol.Xml;

// Create user message
var message = new UserMessage
{
    MessageId = "user-123",
    Contents = new List<AIContent>
    {
        new TextContent { Text = "What is the weather in Seattle?" }
    }
};

var serializer = new MessageSerializer();
var xmlOutput = serializer.Serialize(message);
Console.WriteLine(xmlOutput);

File: snippets/python/user-message_main.py
# from microsoft.agents.xml.models import ChatMessage, TextContent
# from microsoft.agents.xml.serialization import MessageSerializer
#
# # Create user message
# message = ChatMessage(
#     role="user",
#     message_id="user-123",
#     contents=[
#         TextContent(text="What is the weather in Seattle?")
#     ]
# )
#
# serializer = MessageSerializer()
# xml_output = serializer.serialize(message)
# print(xml_output)